File: GoodOfflineGames/GOG.Models/Product.cs
using System.Runtime.Serialization;

using Models.ProductCore;

namespace GOG.Models
{
    [DataContract]
    public class Product: ProductCore
    {
        [DataMember(Name = "customAttributes")]
        public string[] CustomAttributes { get; set; }
        [DataMember(Name = "price")]
        public Price Price { get; set; }
        [DataMember(Name = "isDiscounted")]
        public bool IsDiscounted { get; set; }
        [DataMember(Name = "isInDevelopment")]
        public bool IsInDevelopment { get; set; }
        [DataMember(Name = "releaseDate")]
        public long? ReleaseDate { get; set; }
        [DataMember(Name = "availability")]
        public Availability Availability { get; set; }
        [DataMember(Name = "salesVisibility")]
        public SalesVisibility SalesVisibility { get; set; }
        [DataMember(Name = "buyable")]
        public bool Buyable { get; set; }
        [DataMember(Name = "image")]
        public string Image { get; set; }
        [DataMember(Name = "url")]
        public string Url { get; set; }
        [DataMember(Name = "supportUrl")]
        public string SupportUrl { get; set; }
        [DataMember(Name = "forumUrl")]
        public string ForumUrl { get; set; }
        [DataMember(Name = "worksOn")]
        public WorksOn WorksOn { get; set; }
        [DataMember(Name = "category")]
        public string Category { get; set; }
        [DataMember(Name = "originalCategory")]
        public string OriginalCategory { get; set; }
        [DataMember(Name = "rating")]
        public int Rating { get; set; }
        [DataMember(Name = "type")]
        public int Type { get; set; }
        [DataMember(Name = "isComingSoon")]
        public bool IsComingSoon { get; set; }
        [DataMember(Name = "isPriceVisible")]
        public bool IsPriceVisible { get; set; }
        [DataMember(Name = "isMovie")]
        public bool IsMovie { get; set; }
        [DataMember(Name = "isGame")]
        public bool IsGame { get; set; }
        [DataMember(Name = "slug")]
        public string Slug { get; set; }
    }
}

File: internal/gog/remote/schema/product.go
// Copyright (c) 2020. All rights reserved.
// Use of this source code is governed by an MIT-style license that can be
// found in the LICENSE file.

package schema

type Product struct {
	Developer string   `json:"developer"`
	Publisher string   `json:"publisher"`
	Gallery   []string `json:"gallery"`
	Video     struct {
		ID       string `json:"id"`
		Provider string `json:"provider"`
	} `json:"video"`
	SupportedOperatingSystems []string `json:"supportedOperatingSystems"`
	Genres                    []string `json:"genres"`
	GlobalReleaseDate         int      `json:"globalReleaseDate"`
	IsTBA                     bool     `json:"isTBA"`
	Price                     struct {
		Amount                     string      `json:"amount"`
		BaseAmount                 string      `json:"baseAmount"`
		FinalAmount                string      `json:"finalAmount"`
		IsDiscounted               bool        `json:"isDiscounted"`
		DiscountPercentage         int         `json:"discountPercentage"`
		DiscountDifference         string      `json:"discountDifference"`
		Symbol                     string      `json:"symbol"`
		IsFree                     bool        `json:"isFree"`
		Discount                   int         `json:"discount"`
		IsBonusStoreCreditIncluded bool        `json:"isBonusStoreCreditIncluded"`
		BonusStoreCreditAmount     string      `json:"bonusStoreCreditAmount"`
		PromoID                    interface{} `json:"promoId"`
	} `json:"price"`
	IsDiscounted    bool `json:"isDiscounted"`
	IsInDevelopment bool `json:"isInDevelopment"`
	ID              int  `json:"id" bson:"_id"`
	ReleaseDate     int  `json:"releaseDate"`
	Availability    struct {
		IsAvailable          bool `json:"isAvailable"`
		IsAvailableInAccount bool `json:"isAvailableInAccount"`
	} `json:"availability"`
	SalesVisibility struct {
		IsActive   bool `json:"isActive"`
		FromObject struct {
			Date         string `json:"date"`
			TimezoneType int    `json:"timezone_type"`
			Timezone     string `json:"timezone"`
		} `json:"fromObject"`
		From     int `json:"from"`
		ToObject struct {
			Date         string `json:"date"`
			TimezoneType int    `json:"timezone_type"`
			Timezone     string `json:"timezone"`
		} `json:"toObject"`
		To int `json:"to"`
	} `json:"salesVisibility"`
	Buyable    bool   `json:"buyable"`
	Title      string `json:"title"`
	Image      string `json:"image"`
	URL        string `json:"url"`
	SupportURL string `json:"supportUrl"`
	ForumURL   string `json:"forumUrl"`
	WorksOn    struct {
		Windows bool `json:"Windows"`
		Mac     bool `json:"Mac"`
		Linux   bool `json:"Linux"`
	} `json:"worksOn"`
	Category         string `json:"category"`
	OriginalCategory string `json:"originalCategory"`
	Rating           int    `json:"rating"`
	Type             int    `json:"type"`
	IsComingSoon     bool   `json:"isComingSoon"`
	Slug             string `json:"slug"`
}

// NOTE: Not marshalling the following fields:
// CustomAttributes []interface{} `json:"customAttributes"`
// IsWishlistable   bool   `json:"isWishlistable"`
// IsPriceVisible   bool   `json:"isPriceVisible"`
// IsMovie          bool   `json:"isMovie"`
// IsGame           bool   `json:"isGame"`
